packages/api-derive/src/balances/account.ts

```diff
@@ -84,14 +86,22 @@ function queryCurrent (api: ApiInterfaceRx, accountId: AccountId): Observable<Re
   // AccountInfo is current, support old, eg. Edgeware
   return api.query.system.account<AccountInfo | ITuple<[Index, AccountData]>>(accountId).pipe(
     map((infoOrTuple): Result => {
-      const { feeFrozen, free, miscFrozen, reserved } = (infoOrTuple as AccountInfo).nonce
+      const data = (infoOrTuple as AccountInfo).nonce
         ? (infoOrTuple as AccountInfo).data
         : (infoOrTuple as [Index, AccountData])[1];
 
-      return [
-        (infoOrTuple as AccountInfo).nonce || (infoOrTuple as [Index, AccountData])[0],
-        [[free, reserved, feeFrozen, miscFrozen]]
-      ];
+      const nonce = (infoOrTuple as AccountInfo).nonce || (infoOrTuple as [Index, AccountData])[0];
+
+      if (data) {
```

Comment:
Not 100% sure about this part... with `AccountData = ()` it basically is still an object, in this case the `Null` object since it has not data.

What does work in this case is `if (!data.isEmpty) { /* handle normal case */ } else { /* everything set to zero */ }`